src/bucket/BucketManagerImpl.cpp

```diff
@@ -240,6 +240,48 @@ BucketManagerImpl::forgetUnreferencedBuckets()
         }
     }
 
+    return referenced;
+}
+
+void
+BucketManagerImpl::cleanupStaleFiles()
+{
+    std::lock_guard<std::recursive_mutex> lock(mBucketMutex);
+    auto referenced = getReferencedBuckets();
+    std::transform(std::begin(mSharedBuckets), std::end(mSharedBuckets),
+                   std::inserter(referenced, std::end(referenced)),
```

Comment:
`std::back_inserter` is what you want here

Reply:
It does not work on std::set, so I'm using regular inserter